files/image_config/bootstrap-asic/bootstrap_asic.py

```diff
@@ -0,0 +1,129 @@
+"""
+try:
+    import threading
+    import socket
+    import argparse
+    import os
+    import os.path
+    import time
+    import re
+    from sonic_py_common import daemon_base
+    from swsscommon import swsscommon
+except ImportError as e:
+    raise ImportError(str(e) + " - required module not found")
+
+#
+# Constants ====================================================================
+#
+CHASSIS_ASIC_INFO_TABLE = 'CHASSIS_ASIC_TABLE'
+WAIT_TIMEOUT_SECS = 10
+SELECT_TIMEOUT_MSECS = 5000
+EXTEND_TIMEOUT_USECS = 20000000
+
+# Notifier class to communicate with systemd
+class Notifier():
+    def __init__(self):
+        self.socket = socket.socket(family=socket.AF_UNIX, type=socket.SOCK_DGRAM)
+        self.address = os.getenv("NOTIFY_SOCKET")
+
+    def enabled(self):
+        """Check address set"""
+        return bool(self.address)
+
+    def send(self, msg):
+        """Send message"""
+        self.socket.sendto(msg.encode(), self.address)
+
+    def ready(self):
+        """Notify systemd that this service is ready"""
+        self.send("READY=1\n")
+
+    def status(self, msg):
+        """Notify systemd with status update to display"""
+        self.send("STATUS=%s\n" % (msg,))
+
+    def extend_timeout_usec(self, usec):
+        """Notify systemd that timeout needs to be extended"""
+        self.send("EXTEND_TIMEOUT_USEC=%s\n" %(str(usec)))
+
+class PrintNotifier():
+    def __init__(self):
+        self.print_notifier = 'PrintNotifier'
+
+    def enabled(self):
+        return bool(self.print_notifier)
+
+    def ready(self):
+        print("READY=1\n")
+
+    def status(self, msg):
+        print("STATUS=%s\n" % (msg,))
+
+    def extend_timeout_usec(self, usec):
+        print("EXTEND_TIMEOUT_USEC=%s\n" %(str(usec)))
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("asic", nargs='?', default=0, help="asic instance from 0 to MAX",
+            type=int)
+    args = parser.parse_args()
+
+    stop_event = threading.Event()
+
+    notify = Notifier()
+    #notify = PrintNotifier()
+    if not notify.enabled():
+        raise Exception('Unable to enable systemd notify')
+
+    # If not running on modular chassis supervisor, return success
+    notify.status('Initializing bootstrap-asic service')
+    chassisdb_path='/etc/sonic/chassisdb.conf'
+    if not os.path.isfile(chassisdb_path):
+        notify.ready()
+        notify.status('Skipping bootstrap-asic service')
+        while not stop_event.is_set():
+            time.sleep(WAIT_TIMEOUT_SECS)
+            continue
+
+    # Connect to STATE_DB and subscribe to chassis-module table notifications
+    state_db = daemon_base.db_connect("STATE_DB")
+
+    sel = swsscommon.Select()
+    sst = swsscommon.SubscriberStateTable(state_db, CHASSIS_ASIC_INFO_TABLE)
+    sel.addSelectable(sst)
+
+    asic_done = False
+    while not stop_event.is_set():
+        if asic_done:
+            time.sleep(WAIT_TIMEOUT_SECS)
+            continue
```

Comment:
asic_done is updated to True once the service sees the asic in database. asic_done then never changes. I guess we should update it again to False once asic is gone in database?

Reply:
Please review again. Marking this as resolved since it is outdated.